integration-tests/smoke/ccip_usdc_test.go

```diff
@@ -250,6 +250,134 @@ func TestUSDCTokenTransfer(t *testing.T) {
 	}
 }
 
+func TestMultiSourceUSDCTransfer(t *testing.T) {
+	lggr := logger.TestLogger(t)
+	ctx := ccdeploy.Context(t)
+	tenv, cluster, _ := testsetups.NewLocalDevEnvironmentWithDefaultPrice(t, lggr)
+
+	var endpoint string
+	// When inmemory env then spin up in memory mock server
+	if cluster == nil {
+		server := mockAttestationResponse()
+		defer server.Close()
+		endpoint = server.URL
+	} else {
+		err := actions.SetMockServerWithUSDCAttestation(tenv.Env.MockAdapter, nil)
+		require.NoError(t, err)
+		endpoint = tenv.Env.MockAdapter.InternalEndpoint
+	}
+
+	e := tenv.Env
+	allChainSelectors := maps.Keys(e.Chains)
+	sourceChain1 := allChainSelectors[0]
+	sourceChain2 := allChainSelectors[1]
+	destChain := allChainSelectors[2]
+
```

Comment:
We already have helper functions in test_helpers to take care of this boilerplate code. Please use them.